Documentation/Configuration File.md

```diff
@@ -0,0 +1,36 @@
+# `.sourcekit-lsp` Configuration File
+
+`.sourcekit-lsp` configuration files can be used to modify the behavior of SourceKit-LSP in various ways. The following locations are checked:
+- `~/.sourcekit-lsp/.sourcekit-lsp` (ie. a `.sourcekit-lsp` file in the `~/.sourcekit-lsp` directory)
+- A `.sourcekit-lsp` file in a workspace’s root
```

Comment:
What about `.sourcekit-lsp/config` and require the directory in the workspace root as well? Seems plausible we may add more later. And IMO ideally we'd check `Library/org.swift.sourcekit-lsp/config` for macOS -> `$XDG_CONFIG_HOME/sourcekit-lsp/config` everywhere (though not sure what the standard is for Windows) -> `.sourcekit-lsp/config`.

I do wonder if we need to keep the server options still as well - consider experimental features that may only be supported by specific editors + a user that jumps between editors. Though arguably that case might not work even if you *could* configure them differently. This also makes it more difficult for editors to add support for specific options.

Reply:
Good ideas. I modified the search locations slightly to 

- `~/.sourcekit-lsp/config.json`
- On macOS: `~/Library/Application Support/org.swift.sourcekit-lsp/config.json` from the various `Library` folders on the system
- If the `XDG_CONFIG_HOME` environment variable is set: `$XDG_CONFIG_HOME/org.swift.sourcekit-lsp/config.json`
- A `.sourcekit-lsp/config.json` file in a workspace’s root